packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -400,8 +400,7 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
 
     // The icon size will be scaled by a factor of the accessibility text scale,
     // to follow the behavior of `UISearchTextField`.
-    final double scaledIconSize =
-        MediaQuery.textScaleFactorOf(context) * widget.itemSize;
+    final double scaledIconSize = MediaQuery.textScalerOf(context).textScaleFactor * widget.itemSize;
```

Comment:
What is the plan for migrating things like this off the deprecated textScaleFactor?

Reply:
For this one specifically what I did in the follow up PR is to compute an effective text scale factor to scale the `itemSize` with using the font size of the placeholder text.